services/network/wpa_supplicant.py

```diff
@@ -0,0 +1,868 @@
+            (GO/client).
+        """
+        return self.send_command('P2P-GROUP-REMOVED')
+
+    def send_command_p2p_prov_disc_show_pin(self):
+        """Send message: P2P-PROV-DISC-SHOW-PIN
+
+        P2P_EVENT_PROV_DISC_SHOW_PIN: Request from the peer for us to display a
+            PIN that will be entered on the peer. The following parameters
+            are included after the event prefix: peer_address PIN. The PIN
+            is a random PIN generated for this connection. P2P_CONNECT command
+            can be used to accept the request with the same PIN configured
+            for the connection.
+        """
+        return self.send_command('P2P-PROV-DISC-SHOW-PIN')
+
+    def send_command_p2p_prov_disc_enter_pin(self):
+        """Send message: P2P-PROV-DISC-ENTER-PIN
+
+        P2P_EVENT_PROV_DISC_ENTER_PIN: Request from the peer for us to enter a
+            PIN displayed on the peer. The following parameter is included
+            after the event prefix: peer address.
+        """
+        return self.send_command('P2P-PROV-DISC-ENTER-PIN')
+
+    def send_command_p2p_prov_disc_pbc_req(self):
+        """Send message: P2P-PROV-DISC-PBC-REQ
+
+        P2P_EVENT_PROV_DISC_PBC_REQ: Request from the peer for us to connect using
+            PBC. The following parameters are included after the event prefix:
+            peer_address. P2P_CONNECT command can be used to accept the request.
+
+        """
+        return self.send_command('P2P-PROV-DISC-PBC-REQ')
+
+    def send_command_p2p_prov_disc_pbc_resp(self):
+        """Send message: P2P-PROV-DISC-PBC-RESP
+
+        P2P-SERV-DISC-RESP: Indicate reception of a P2P service discovery response.
+            The following parameters are included after the event prefix: source
+            address, Service Update Indicator, Service Response TLV(s) as hexdump.
+
+        """
+        return self.send_command('P2P-PROV-DISC-PBC-RESP')
+
+    def send_command_p2p_invitation_received(self):
+        """Send message: P2P-INVITATION-RECEIVED
+
+        P2P-INVITATION-RECEIVED: Indicate reception of a P2P Invitation Request.
+            For persistent groups, the parameter after the event prefix indicates
+            which network block includes the persistent group data.
+        """
+        return self.send_command('P2P-INVITATION-RECEIVED')
+
+    def send_command_p2p_invitation_result(self):
+        """Send message: P2P-INVITATION-RESULT
+
+        shows the status code returned by the peer (or -1 on local failure or timeout).
+
+        """
+        return self.send_command('P2P-INVITATION-RESULT')
+
+
+if __name__ == "__main__":
+    wpa = WPASupplicant()
+    wpa.run(('0.0.0.0', 6664))
+    time.sleep(1)
+    wpa.send_command_list_networks()
+    for i in range(5):
+        wpa.send_command_remove_network(str(i))
+
+    wpa.send_command_add_network()
+    wpa.send_command_set_network('0', 'ssid', '\"wifi_ssid\"')
```

Comment:
this is odd.. why do you need to send the quotes?

Reply:
ask wpa_supplicant api, that's how it works.